src/main/java/org/cloudburstmc/api/util/TextFormat.java

```diff
@@ -0,0 +1,286 @@
+package org.cloudburstmc.api.util;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.regex.Pattern;
+
+/**
+ * All supported formatting values for chat and console.
+ */
+public enum TextFormat {
+    /**
+     * Represents black.
+     */
+    BLACK('0', 0x00),
+    /**
+     * Represents dark blue.
+     */
+    DARK_BLUE('1', 0x1),
+    /**
+     * Represents dark green.
+     */
+    DARK_GREEN('2', 0x2),
+    /**
+     * Represents dark blue (aqua).
+     */
+    DARK_AQUA('3', 0x3),
+    /**
+     * Represents dark red.
+     */
+    DARK_RED('4', 0x4),
+    /**
+     * Represents dark purple.
+     */
+    DARK_PURPLE('5', 0x5),
+    /**
+     * Represents gold.
+     */
+    GOLD('6', 0x6),
+    /**
+     * Represents gray.
+     */
+    GRAY('7', 0x7),
+    /**
+     * Represents dark gray.
+     */
+    DARK_GRAY('8', 0x8),
+    /**
+     * Represents blue.
+     */
+    BLUE('9', 0x9),
+    /**
+     * Represents green.
+     */
+    GREEN('a', 0xA),
+    /**
+     * Represents aqua.
+     */
+    AQUA('b', 0xB),
+    /**
+     * Represents red.
+     */
+    RED('c', 0xC),
+    /**
+     * Represents light purple.
+     */
+    LIGHT_PURPLE('d', 0xD),
+    /**
+     * Represents yellow.
+     */
+    YELLOW('e', 0xE),
+    /**
+     * Represents white.
+     */
+    WHITE('f', 0xF),
+    /**
+     * Represents minecoin gold.
+     */
+    MINECOIN_GOLD('g', 0x16),
+    /**
+     * Makes the text obfuscated.
+     */
+    OBFUSCATED('k', 0x10, true),
+    /**
+     * Makes the text bold.
+     */
+    BOLD('l', 0x11, true),
+    /**
+     * Makes a line appear through the text.
+     */
+    STRIKETHROUGH('m', 0x12, true),
+    /**
+     * Makes the text appear underlined.
+     */
+    UNDERLINE('n', 0x13, true),
+    /**
+     * Makes the text italic.
+     */
+    ITALIC('o', 0x14, true),
+    /**
+     * Resets all previous chat colors or formats.
+     */
+    RESET('r', 0x15);
+
+    /**
+     * The special character which prefixes all format codes. Use this if
+     * you need to dynamically convert format codes from your custom format.
+     */
+    public static final char ESCAPE = '\u00A7';
+
+    private static final Pattern CLEAN_PATTERN = Pattern.compile("(?i)" + ESCAPE + "[0-9A-GK-OR]");
+    private final static Map<Integer, TextFormat> BY_ID = new TreeMap<>();
```

Comment:
I'm not sure why this is a `TreeMap` when the below field is `HashMap`, but I removed the usage of guava's `Maps.newTreeMap()` here anyway.